### PR TITLE
Update 6 custom resources for Chef 13

### LINKED ISSUE #70: "uninitialized constant" error with chef-client 13.X run
### Cookbook version
1.2.1

### Chef-client version
13.0.118

### Platform Details
aix "7100-04-04-1717"

### Scenario:
Trying to run the cookbook with a simple recipe as below (issue happens on all different recipes)
aix_tcpservice 'xntpd' do
  action :enable
end

### Steps to Reproduce:
run the above recipe (or infant any recipe)

### Expected Result:
expecting the xntpd service to be enabled in /etc/rc.tcpip

### Actual Result: 
[chef-stacktrace.txt](https://github.com/chef-cookbooks/aix/files/985739/chef-stacktrace.txt)

Starting Chef Client, version 13.0.118
resolving cookbooks for run list: ["aix"]
Synchronizing Cookbooks:
  - aix (1.2.1)
Installing Cookbook Gems:
Compiling Cookbooks...
/etc/chef/local-mode-cache/cache/cookbooks/aix/libraries/helpers.rb:357: warning: key "xpm" is duplicated and overwritten on line 358
Converging 1 resources
Recipe: aix::default
  * aix_tcpservice[xntpd] action enable

    ================================================================================
    Error executing action `enable` on resource 'aix_tcpservice[xntpd]'
    ================================================================================

    NameError
    ---------
    uninitialized constant Chef::Resource::AixTcpservice

    Cookbook Trace:
    ---------------
    /etc/chef/local-mode-cache/cache/cookbooks/aix/providers/tcpservice.rb:25:in `load_current_resource'

    Resource Declaration:
    ---------------------
    # In /etc/chef/local-mode-cache/cache/cookbooks/aix/recipes/default.rb

      1: aix_tcpservice 'xntpd' do
      2:   action :enable
      3: end

    Compiled Resource:
    ------------------
    # Declared in /etc/chef/local-mode-cache/cache/cookbooks/aix/recipes/default.rb:1:in `from_file'

    aix_tcpservice("xntpd") do
      action [:enable]
      default_guard_interpreter :default
      declared_type :aix_tcpservice
      cookbook_name "aix"
      recipe_name "default"
      identifier "xntpd"
    end

    System Info:
    ------------
    chef_version=13.0.118
    platform=aix
    platform_version=7.1
    ruby=ruby 2.4.1p111 (2017-03-22 revision 58053) [powerpc-aix6.1.0.0]
    program_name=chef-client worker: ppid=11272326;start=02:33:10;
    executable=/opt/chef/bin/chef-client

Running handlers:
[2017-05-09T02:33:18-04:00] ERROR: Running exception handlers
[2017-05-09T02:33:18-04:00] ERROR: Running exception handlers
Running handlers complete
[2017-05-09T02:33:18-04:00] ERROR: Exception handlers complete
[2017-05-09T02:33:18-04:00] ERROR: Exception handlers complete
Chef Client failed. 0 resources updated in 08 seconds
[2017-05-09T02:33:18-04:00] FATAL: Stacktrace dumped to /etc/chef/local-mode-cache/cache/chef-stacktrace.out
[2017-05-09T02:33:18-04:00] FATAL: Stacktrace dumped to /etc/chef/local-mode-cache/cache/chef-stacktrace.out
[2017-05-09T02:33:18-04:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2017-05-09T02:33:18-04:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2017-05-09T02:33:18-04:00] ERROR: aix_tcpservice[xntpd] (aix::default line 1) had an error: NameError: uninitialized constant Chef::Resource::AixTcpservice
[2017-05-09T02:33:18-04:00] ERROR: aix_tcpservice[xntpd] (aix::default line 1) had an error: NameError: uninitialized constant Chef::Resource::AixTcpservice
[2017-05-09T02:33:18-04:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
[2017-05-09T02:33:18-04:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

### PR DESCRIPTION
**Pull request by msgarbossa**
### Description

Removes provider files and replaces custom resources with ones supported in Chef 13 for the following custom resources:
- chdev
- chsec
- no
- subserver
- tcpservice
- tunables

### Issues Resolved

Resolves #70 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: resources/chdev.rb [hunk 1]
```diff
@@ -14,12 +14,78 @@
 # limitations under the License.
 #
 
-actions :update
+resource_name :aix_chdev
```
Comment (tas50) [inline resources/chdev.rb]: This isn't needed since the auto generated name will be the same

File: resources/chdev.rb [hunk 2]
```diff
@@ -14,12 +14,78 @@
...
+
+property :name, String, name_property: true, identity: true
```
Comment (tas50) [inline resources/chdev.rb]: You don't actually want to set the name property. It comes along already

File: resources/chdev.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright:: 2015-2016, Benoit Creau <benoit.creau@chmod666.org>
```
Comment (tas50) [inline resources/chdev.rb]: All the existing copyrights should be retained

File: resources/subserver.rb [hunk 4]
```diff
@@ -14,20 +14,86 @@
...
-attribute :user, kind_of: String, default: 'root', required: true
-attribute :program, kind_of: String
-attribute :args, kind_of: String
-
-attr_accessor :enabled
+property :servicename, String, name_property: true, identity: true
+property :type, String, equal_to: %w[dgram stream sunrpc_udp sunrpc_tcp]
+property :protocol, String, required: true, equal_to: %w[tcp udp tcp6 udp6], identity: true
+property :wait, String, equal_to: %w[wait nowait SRC], default: 'nowait'
+property :user, String, required: true, default: 'nowait'
```
Comment (tas50) [inline resources/subserver.rb]: Looks like the default value here changed on accident

File: resources/subserver.rb [hunk 5]
```diff
@@ -14,20 +14,86 @@
...
+property :program, String
+property :args, String
+property :enabled, [true, false]
```
Comment (tas50) [inline resources/subserver.rb]: If you're determining the state of these within the cookbook you need to set desired_state: false

File: resources/tcpservice.rb [hunk 6]
```diff
@@ -14,11 +14,54 @@
...
+resource_name :aix_tcpservice
 
-attribute :identifier, name_attribute: true, kind_of: String
-attribute :immediate, kind_of: [TrueClass, FalseClass], default: false
-
-attr_accessor :enabled
+property :identifier, String, name_property: true
+property :immediate, [true, false], default: false
 
 default_action :enable
```
Comment (tas50) [inline resources/tcpservice.rb]: default_action is the 1st action in a custom resource

File: metadata.rb [hunk 7]
```diff
@@ -26,4 +26,4 @@
 
 supports 'aix', '>= 6.1'
 
-chef_version '>= 12.1'
+chef_version '>= 12.5'
```
Comment (tas50) [inline metadata.rb]: You actually want to set this to 12.7 since 12.5 and 12.6 had some funky custom resource bugs.

**Comment by tas50** [comment]
I'm pretty stoked to see this PR come in. This gets us headed in the right direction with these resources. Thanks for taking the time to put it together. I've left some feedback on some custom resource-isms. I can't really speak to the AIX nature of the cookbook, since I've never actually used AIX, so I'll have to trust that you've test this somewhere.

**Comment by tas50** [comment]
Also there's some cookstyle warnings that you can probably mostly autocorrect

File: metadata.rb (hunk 7 above)
Comment (tas50) [reply metadata.rb]: Also make sure to update the version in the readme

**Comment by msgarbossa** [comment]
Thanks for the feedback.  The latest commits should cover everything you pointed out.  I have places that I tested the changes.  I tried not to change anymore than needed to get this to work on Chef 13 so hopefully this doesn't introduce regressions.  The previous custom resources weren't very useful for us since we had to move to Chef 13 to resolve other issues.

**Comment by tas50** [comment]
I'll fix up those cookstyle warnings on master. Thanks for taking this on. A lot of people are going to be really happy with this

**Pull request merged**